plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -202,7 +202,7 @@ private void evaluateFacetConfiguration() {
   private void updatePageBasedOnFacetConfig(boolean hasCorrectFacetConfiguration) {
     if (hasCorrectFacetConfiguration) {
       canSetMessage = true;
-      setErrorMessage(null);
+      content.getDataBindingContext().updateTargets();
```

Comment:
Thanks @chanseokoh I was trying to figure out the reason for the logic, but I missed this. I'm now trying to figure out what was fixed, because I still can see the error mentioned in https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/880#issuecomment-256149028 :
 - checkout `master` and launch the plugin
 - open properties of a standard project
 - open the deploy preferences
 - remove standard and add flex facet
 - go back to the deploy preferences page
 - type some illegal version name (e.g.: `#`)

The `Apply` and `OK` buttons disable, no error message is shown regarding the version, only that this page is valid for standard projects only. When I try to switch to another page, I get the message that there are invalid values on the page I cannot switch.